modules/local/mirtop_quant.nf

```diff
@@ -24,9 +24,12 @@ process MIRTOP_QUANT {
     script:
     def filter_species = params.mirgenedb ? params.mirgenedb_species : params.mirtrace_species
     """
-    mirtop gff --hairpin $hairpin --gtf $gtf -o mirtop --sps $filter_species ./bams/*
-    mirtop counts --hairpin $hairpin --gtf $gtf -o mirtop --sps $filter_species --add-extra --gff mirtop/mirtop.gff
-    mirtop export --format isomir --hairpin $hairpin --gtf $gtf --sps $filter_species -o mirtop mirtop/mirtop.gff
+    #Cleanup the GTF if mirbase html form is broken
+    GTF="$gtf"
+    sed 's/&gt;/>/g' \$GTF | sed 's#<br>#\\n#g' | sed 's#</p>##g' | sed 's#<p>##g' > \${GTF}_html_cleaned.fa
```

Comment:
So this is the magic fix! is there a reason the gff is renamed to {GTF}_html_cleaned.fa?

Reply:
So when running with `-profile test_full`, this gives the following error:

```python
Traceback (most recent call last):
['gff', '--hairpin', 'hairpin.fa_igenome.fa_idx.fa', '--gtf', 'hsa.gff3_html_cleaned.fa', '-o', 'mirtop', '--sps', 'hsa', './bams/c1_seqcluster.bam', './bams/c2_seqcluster.bam', './bams/c3_seqcluster.bam', './bams/c4_seqcluster.bam', './bams/c5_seqcluster.bam', './bams/c6_seqcluster.bam', './bams/c7_seqcluster.bam', './bams/cc1_seqcluster.bam', './bams/cc2_seqcluster.bam', './bams/cc3_seqcluster.bam', './bams/cc4_seqcluster.bam', './bams/cc5_seqcluster.bam', './bams/cc6_seqcluster.bam', './bams/cc7_seqcluster.bam', './bams/cpc1_seqcluster.bam', './bams/cpc2_seqcluster.bam', './bams/cpc3_seqcluster.bam', './bams/cpc4_seqcluster.bam', './bams/cpc5_seqcluster.bam', './bams/cpc6_seqcluster.bam', './bams/cpc7_seqcluster.bam', './bams/pc1_seqcluster.bam', './bams/pc2_seqcluster.bam', './bams/pc3_seqcluster.bam', './bams/pc4_seqcluster.bam', './bams/pc5_seqcluster.bam', './bams/pc6_seqcluster.bam', './bams/pc7_seqcluster.bam']
  File "/usr/local/bin/mirtop", line 10, in
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/mirtop/command_line.py", line 31, in main
    reader(kwargs["args"])
  File "/usr/local/lib/python3.9/site-packages/mirtop/gff/__init__.py", line 28, in reader
    matures = mapper.read_gtf_to_precursor(args.gtf)
  File "/usr/local/lib/python3.9/site-packages/mirtop/mirna/mapper.py", line 165, in read_gtf_to_precursor
    mapped = read_gtf_to_precursor_mirbase(gtf)
  File "/usr/local/lib/python3.9/site-packages/mirtop/mirna/mapper.py", line 332, in read_gtf_to_precursor_mirbase
    chrom, start, end, strand = cols[0], cols[3], cols[4], cols[6]
IndexError: list index out of range
```

I tested this locally to see what was happening, and it looks like the sed stuff adds an extra line into the gff file, which doesn't make mirtop happy, because it doesn't seem to have any logic for dealing with empty lines, and will just panic if it sees an empty line (I think in this case it is just a solo newline character??) https://github.com/miRTop/mirtop/blob/170de820a2c83c09223424def68261660906ae55/mirtop/mirna/mapper.py#L323-L333

So a fix would be to just do some additional bash trickery to drop the blank line. 
Something along these lines seemed to work for me:
```bash
echo "$(echo "$(tac hsa.gff3_html_cleaned.fa)" | tac)" > hsa.gff3_html_cleaned.gff3
```

I'm far from a bash wizard though, I stumbled upon that solution here (https://stackoverflow.com/questions/7359527/removing-trailing-starting-newlines-with-sed-awk-tr-and-friends). I'm sure any other sed thing would work too.

Reply:
Adding this here as SED magic to make mirtop happy

`sed -e :a -e '/^\n*$/{$d;N;};/\n$/ba'` https://stackoverflow.com/questions/7359527/removing-trailing-starting-newlines-with-sed-awk-tr-and-friends